test/unit/org/apache/cassandra/cql3/RandomSchemaTest.java

```diff
@@ -0,0 +1,242 @@
+            int primaryColumnCount = primaryColumnCount(metadata);
+            String selectStmt = selectStmt(metadata);
+
+            for (int i = 0; i < 100; i++)
+            {
+                ByteBuffer[] expected = dataGen.generate(random);
+                try
+                {
+                    ByteBuffer[] rowKey = Arrays.copyOf(expected, primaryColumnCount);
+                    execute(insertStmt, expected);
+                    // check memtable
+                    assertRows(execute(selectStmt, rowKey), expected);
+                    assertRowsNet(executeNet(selectStmt, rowKey), expected);
+
+                    // check sstable
+                    flush(KEYSPACE, metadata.name);
+                    compact(KEYSPACE, metadata.name);
+                    assertRows(execute(selectStmt, rowKey), expected);
+                    assertRowsNet(executeNet(selectStmt, rowKey), expected);
+
+                    execute("TRUNCATE " + metadata);
+                }
+                catch (Throwable t)
+                {
+                    Iterator<ColumnMetadata> it = metadata.allColumnsInSelectOrder();
+                    List<String> literals = new ArrayList<>(expected.length);
+                    for (int idx = 0; idx < expected.length; idx++)
+                    {
+                        assert it.hasNext();
+                        ColumnMetadata meta = it.next();
+                        literals.add(!expected[idx].hasRemaining() ? "empty" : meta.type.asCQL3Type().toCQLLiteral(expected[idx]));
+                    }
+                    AssertionError error = new AssertionError(String.format("Failure at attempt %d with schema\n%s\nfor values %s", i, createTable, literals), t);
+                    throw error;
+                }
+            }
+        });
+    }
+
+    private void maybeCreateUDTs(TableMetadata metadata)
+    {
+        Set<UserType> udts = CassandraGenerators.extractUDTs(metadata);
+        if (!udts.isEmpty())
+        {
+            Deque<UserType> pending = new ArrayDeque<>();
+            pending.addAll(udts);
+            Set<ByteBuffer> created = new HashSet<>();
+            while (!pending.isEmpty())
+            {
+                UserType next = pending.poll();
+                Set<UserType> subTypes = AbstractTypeGenerators.extractUDTs(next);
+                subTypes.remove(next); // it includes self
+                if (subTypes.isEmpty() || subTypes.stream().allMatch(t -> created.contains(t.name)))
+                {
+                    String cql = next.toCqlString(false, false);
+                    logger.warn("Creating UDT {}", cql);
+                    schemaChange(cql);
+                    created.add(next.name);
+                }
+                else
+                {
+                    logger.warn("Unable to create UDT {}; following sub-types still not created: {}", next.getCqlTypeName(), subTypes.stream().filter(t -> !created.contains(t.name)).collect(Collectors.toSet()));
```

Comment:
We can break the long line:
```suggestion
                    logger.warn("Unable to create UDT {}; following sub-types still not created: {}",
                                next.getCqlTypeName(),
                                subTypes.stream().filter(t -> !created.contains(t.name)).collect(Collectors.toSet()));
```